### PR TITLE
dev-libs/libedit: bump to 20170329.3.1

### PR DESCRIPTION
**Pull request by yegortimoshenko**
There are quite a few changes (especially to `el_fn_sh_complete` patch). It'd be great to upstream it, has anyone tried that? I've sent smaller patches to the developer but not this one (I'm not yet sure it works properly).

Could someone please try this version out with Gentoo/kFreeBSD `sh(1)`?

### CODE REVIEW

File: dev-libs/libedit/libedit-20170329.3.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+EAPI="6"
+inherit eutils flag-o-matic toolchain-funcs versionator multilib-minimal
+
+MY_PV=$(get_major_version)-$(get_after_major_version)
+MY_P=${PN}-${MY_PV}
+
+DESCRIPTION="BSD replacement for libreadline"
+HOMEPAGE="http://www.thrysoee.dk/editline/"
+SRC_URI="${HOMEPAGE}${MY_P}.tar.gz"
```
Comment (mgorny) [inline dev-libs/libedit/libedit-20170329.3.1.ebuild]: Please don't use HOMEPAGE here. By definition, HOMEPAGE can list multiple sites and reusing it as a single value is contrary to the principle of least surprise ;-).

File: dev-libs/libedit/libedit-20170329.3.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="6"
+inherit eutils flag-o-matic toolchain-funcs versionator multilib-minimal
+
+MY_PV=$(get_major_version)-$(get_after_major_version)
```
Comment (mgorny) [inline dev-libs/libedit/libedit-20170329.3.1.ebuild]: MY_PV=${PV/./-}

would be equivalent, wouldn't it? If so, you can avoid using versionator (i.e. save code).

File: dev-libs/libedit/libedit-20170329.3.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+multilib_src_install() {
+	emake DESTDIR="${D}" install
+	gen_usr_ldscript -a edit
+}
+
+multilib_src_install_all() {
+	einstalldocs
+	prune_libtool_files --all
```
Comment (mgorny) [inline dev-libs/libedit/libedit-20170329.3.1.ebuild]: find "${D}" -name '*.la' -delete || die

is how we do it these days. It should also save dep on ltprune (eutils previously).

**Comment by yegortimoshenko** [comment]
Done. Thanks for the tips!

**Comment by nigoro** [comment]
@yegortimoshenko, Your BSD patch works fine. Thanks for update.
But, I got file collision error on Gentoo/Linux and Gentoo/FreeBSD.
Would you fix it ?

```
>>> Installing (1 of 1) dev-libs/libedit-20170329.3.1::gentoo
<snip>
 * Detected file collision(s):
 *
 *      /usr/share/man/man3/history.3.bz2
 *
 * Searching all installed packages for file collisions...
 *
 * Press Ctrl-C to Stop
 *
 * sys-libs/readline-7.0_p3:0::gentoo
 *      /usr/share/man/man3/history.3.bz2
```

Steps to reproduce)
emerge -C dev-libs/libedit
emerge sys-libs/readline
emerge dev-libs/libedit

**Comment by yegortimoshenko** [comment]
@nigoro Thank you so much for testing this PR on Gentoo/FreeBSD!

This seems to be an issue with previous versions of `dev-libs/libedit` as well. Can't think of a good way to solve that won't require a `sys-libs/readline` ebuild update.

**Comment by mgorny** [comment]
I think we can just remove it from libedit for now, since readline is the 'canonical' provider for now.

**Comment by nigoro** [comment]
Yeah, I agree with @mgorny.
In addition, history.3.bz2 is just a symbolic link to editline.3.bz2.
I think it's no problem to remove it.
```
# cd /var/tmp/portage/dev-libs/libedit-20170329.3.1/image/usr/share/man/man3 && ls -la history.3.bz2
lrwxr-xr-x  1 root  portage  14 Jun 11 20:38 history.3.bz2 -> editline.3.bz2
````

@mgorny, I know you are llvm team leader.
FYI, if latest libedit merge into main tree, probably sys-devel/lldb with USE="libedit" won't be worked correctly like other distributions.
Details, please see upstream bugzilla (it's already fixed in trunk).
https://bugs.llvm.org/show_bug.cgi?id=28898

**Comment by mgorny** [comment]
@nigoro, thanks for the heads up. I suppose I'll just backport the fix to 4.0 then.

**Comment by mgorny** [comment]
Fixed the file collision and a few minor issues, added a patch to support `ncurses[tinfo]`, merged it and backported the fix to `dev-util/lldb-4.0.0-r1`.

**Pull request closed without merging**